dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -432,3 +614,68 @@ def dpnp_solve(a, b):
         a_ht_copy_ev.wait()
 
         return b_f
+
+
+def dpnp_slogdet(a):
+    """
+    dpnp_slogdet(a)
+
+    Returns sign and logarithm of the determinant of `a` array.
+
+    """
+
+    a_usm_type = a.usm_type
+    a_sycl_queue = a.sycl_queue
+
+    res_type = _common_type(a)
+    logdet_dtype = dpnp.dtype(res_type.char.lower())
```

Comment:
It looks a bit tricky, we fully rely here that complex to appropriate real dtype can be casted in numpy by lowering the char, like "F" -> 'f" to map from numpy.complex64 to numpy.float32.
I believe it would be good to leave a comment at least or to implement an explicit mapping, like numpy did.